code/mob/living/life/stuns_lying.dm

```diff
@@ -65,7 +65,8 @@
 
 		if (owner.lying != lying_old)
 			owner.update_lying()
-			owner.set_density(!owner.lying)
+			if(!(owner.mob_flags & NO_DENSITY))
+				owner.set_density(!owner.lying)
```

Comment:
couldn't this be handled *in* the set_density proc?

Reply:
That's how I did it at first, but I felt that doing mob-specific stuff inside `set_density` (a generalized proc that ideally shouldn't really care what kind of subtype it's called on) was kind of smelly. I could just make it a regular flag too and do it inside `set_density` but it's only really needed for things that tend to jump back and forth between different density values, which is mostly just mobs iirc.

Reply:
Yeah that's a better idea